### ISSUE TITLE
Exception when constructing proxied class

### ISSUE THREAD

**Issue by sethbrenith** [Microsoft/ChakraCore#5225]
Minimum repro:

```javascript
class A {}
const p = new Proxy(A, {});
Reflect.construct(p, [], A);
```

My VSO branch user/sethb/bound-class-2 has a bunch of new test cases based on suggestions in #5218. I think all of the failures boil down to this same problem. Reflect.construct passes new.target as an extra arg on the end, but the class constructor expects new.target as arg zero. Proxy probably needs to be responsible for putting the argument in the correct place when calling a class constructor. Current behavior is that the class constructor tries to get function info about arg zero (some newly-initialized object), and throws "TypeError: Object doesn't support this action".

**Comment by jdalton** [comment]
I'm seeing this error in node-chakracore 10.6.0 when testing for [this bug](https://github.com/Microsoft/ChakraCore/issues/4729).

```js
var dummyProxy = new Proxy(class P {}, {})
class C extends dummyProxy { c() {} }
new C().c !== void 0 // throws the action not supported error
```

**Comment by sethbrenith** [comment]
@jdalton good point. I think I need to port this fix from master to 1.10 so it can show up in Node builds.
